python/foglamp/services/core/api/service.py

```diff
@@ -95,42 +96,51 @@ def get_service_records():
         is_enabled = True if ((type(enabled) is str and enabled.lower() in ['t', 'true']) or (
             (type(enabled) is bool and enabled is True))) else False
 
-        storage = connect.get_storage_async()
+        # ---------------------------------------- Sanity checks
+        # Check if a valid plugin has been provided
+        try:
+            # "plugin_module_path" is fixed by design. It is MANDATORY to keep the plugin in the exactly similar named
+            # folder, within the plugin_module_path.
+            plugin_module_path = "foglamp.plugins.south" if service_type == 'south' else \
+                                 "foglamp.plugins.north" if service_type == 'north' else None
+            import_file_name = "{path}.{dir}.{file}".format(path=plugin_module_path, dir=plugin, file=plugin)
+            _plugin = __import__(import_file_name, fromlist=[''])
+        except ImportError as ex:
+            raise web.HTTPInternalServerError(reason='Plugin "{}" import problem from path "{}". {}'.format(plugin, plugin_module_path, str(ex)))
+        except Exception as ex:
```

Comment:
we should check import error ONLY, this should be the parent exception block if any;